pxteditor/projectImport.ts

```diff
@@ -0,0 +1,49 @@
+let _db: pxt.BrowserUtils.IDBWrapper;
+
+
+interface StoredProject {
+    importId: string;
+    project: pxt.workspace.Project;
+}
+
+const PROJECT_TABLE = "projects";
+const KEYPATH = "importId";
+
+export async function saveProjectAsync(project: pxt.workspace.Project): Promise<string> {
+    const toStore: StoredProject = {
+        importId: pxt.U.guidGen(),
+        project
+    };
+
+    const db = await initAsync();
+    await db.deleteAllAsync(PROJECT_TABLE);
+    await db.setAsync(PROJECT_TABLE, toStore);
+
+    return toStore.importId;
+}
+
+export async function removeProjectAsync(importId: string): Promise<pxt.workspace.Project> {
+    const db = await initAsync();
+
+    const stored = await db.getAsync<StoredProject>(PROJECT_TABLE, importId);
+    await db.deleteAllAsync(PROJECT_TABLE);
+
+    return stored.project;
```

Comment:
I think this should return null or undefined if `stored` doesn't exist, otherwise we error on `stored.project` rather than running the error handling in importEmbedProjectAsync (which checks project is truthy).